app/teachingCallResponseReport/services/teachingCallResponseReportStateService.js

```diff
@@ -236,77 +236,8 @@ class TeachingCallResponseReportStateService {
 			}
 		};	
 	}
-
-	/**
-	 * @param  {array} blob A 75 length array representing unavailabilities
-	 */
-	availabilityBlobToDescriptions (blob) {
-		var self = this;
-		var hoursArray = blob.split(',');
-
-		if (hoursArray.length != 75) {
-			return null;
-		}
-
-		var descriptions = [];
-		descriptions.push(self.describeDayArray(hoursArray.slice(0,14), "M"));
-		descriptions.push(self.describeDayArray(hoursArray.slice(15,29), "T"));
-		descriptions.push(self.describeDayArray(hoursArray.slice(30,44), "W"));
-		descriptions.push(self.describeDayArray(hoursArray.slice(45,59), "R"));
-		descriptions.push(self.describeDayArray(hoursArray.slice(60,74), "F"));
-
-		return descriptions;
-	}
-
-	describeDayArray (dayArray, dayCode) {
-		var self = this;
-		var descriptions = {
-			day: dayCode,
-			times: ""
-		};
-
-		var startHour = 7;
-
-		var startTimeBlock = null;
-		var endTimeBlock = null;
-		var blocks = [];
-
-		dayArray.forEach( function(hourFlag, i) {
-			if (hourFlag == "1") {
-				if (startTimeBlock == null) {
-					startTimeBlock = startHour + i;
-					endTimeBlock = startHour + i + 1;
-				} else {
-					endTimeBlock++;
-				}
-			} else if (hourFlag == "0" && startTimeBlock != null) {
-				blocks.push(self.blockDescription(startTimeBlock, endTimeBlock));
-				startTimeBlock = null;
-			}
-		});
-
-		if (startTimeBlock != null) {
-			blocks.push(self.blockDescription(startTimeBlock, endTimeBlock));
-		}
-
-		if(blocks.length == 0) {
-			// No availabilities were indicated
-			blocks.push("Not available");
-		}
-
-		descriptions.times = blocks.join(", ");
-
-		return descriptions;
-	}
-
-	blockDescription (startTime, endTime) {
-		var start = (startTime > 12 ? (startTime - 12) + "pm" : startTime + "am" );
-		var end = (endTime > 12 ? (endTime - 12) + "pm" : endTime + "am" );
-
-		return start + "-" + end;
-	}
 }
 
-TeachingCallResponseReportStateService.$inject = ['$rootScope', '$log', 'Term', 'SectionGroup', 'ActionTypes', 'StringService'];
+TeachingCallResponseReportStateService.$inject = ['$rootScope', '$log', 'Term', 'SectionGroup', 'ActionTypes', 'StringService', 'SupportCallService'];
```

Comment:
I appreciate you not repeating the blockDescription code but simply putting it in SupportCallService and including that service here is not the right thing to do. It is not inherently a "SupportCallService" method.

Please place it somewhere outside of both methods, perhaps in a helper?